kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/PlainSaslServer.java

```diff
@@ -81,7 +83,14 @@ public String getMechanismName() {
             if (proxyRoles != null && proxyRoles.contains(authState.getAuthRole())) {
                 // the Proxy passes the OriginalPrincipal as "username"
                 authorizationId = saslAuth.getUsername();
-                log.info("Authenticated Proxy role {} as user role {}", authState.getAuthRole(), authorizationId);
+                username = null; // PULSAR TENANT
+                if (authorizationId.contains("/")) {
+                    int lastSlash = authorizationId.lastIndexOf('/');
+                    username = authorizationId.substring(lastSlash + 1);
```

Comment:
this is possible.
if you enable "authorisation" this is not a problem because the authenticated user must be able to access the tenant.
if you do not enable "authentication" then we are not using this feature (because there is no "username")

if you only enable "authentication" and not "authorisation" then we have the problem you are saying.

I have two ideas:
1. Disable automatic tenant creation in case the user does not enable authorisation.
2. Do not create the tenant if the tenant is NOT `kafkaMetadataTenant`

Reply:
I have a concern, when user login with illegal tenant and use the hack way to call the `handleFindCoordinatorRequest`, because the authorization not check on `handleFindCoordinatorRequest`, may have the problem I say.

Reply:
we must add an authorization if it is missing in all the calls that require access to a coordinator.

I have to check again I thought we had it.
I am working on the proxy and in the proxy I am adding more authn checks and I missed to add them here.

**Is it okay for you if I add the authz check in every place we access a coordinator ?**

Reply:
We can't add authorization check in `handleFindCoordinatorRequest`, because we don't have topic name or namespace in request, Pulsar authorization depend on topic name or namespace.

If I understand correct, you means check authorization when get coordinators?

Reply:
I have changed my mind.
I will add authz check during authentication. it is easier and we are sure that we are covering all cases

Reply:
I have implemented the check during authentication.
Unfortunately there is no concept of tenant level permissions that is not being an admin,
so actually I will just test that the tenant exists.
it is enough in order to prevent problems like the ones you described.

Reply:
we have a problem in adding some check that the user actually can access the tenant, because we need to access `tenant/__kafka/__consumer_offsets`

but we are not sure that the user has access to `tenant/__kafka`, because it is a system topic.

Reply:
I can address this problem in a follow up patch

Reply:
> I can address this problem in a follow up patch
Great !

Reply:
this is true, but I have this case:
I have a JWT token that is valid for tenant **foo**
The user logs in as **bar**:
- he will be able to access only topics for `foo` namespace
- but it will it will use `bar/__kafka` metadata namespace
 
This is a problem because in a multi tenant system the administrator wants that only **bar** users (people that have access to tenant *bar*) can interfere with the `bar` tenant.
 
My understanding is that in Pulsar you do not have tenant level permissions (apart from "admin"), so in my code I have an assertions about the ability to access the system topics (bar/__kafka/__consumer_offsets...) using the authenticated principal.

This is probably something that must be configurable because in Kafka model the user is not required to have access to the metadata topics.

I believe that this is something we should address in a follow up work, because it depends on the way the administrator assigns tokens to the users.

Reply:
I understand what you said now.

If we force user login use `tenant/namespace`, then we can avoid this problem, but this will limit users access other `namespace`. I don't know users will cross `namespace` produce and consume in same tenant or not.

If we have `tenant/namespace` then we can use follow code to check `tenant/namespace` is exists.
```java
getPulsarService().getPulsarResources().getNamespaceResources()
```